darwin/Platform.c

```diff
@@ -292,13 +292,28 @@ double Platform_setCPUValues(Meter* mtr, unsigned int cpu) {
 
 void Platform_setMemoryValues(Meter* mtr) {
    const DarwinMachine* dhost = (const DarwinMachine*) mtr->host;
+#ifdef HAVE_STRUCT_VM_STATISTICS64
+   const struct vm_statistics64* vm = &dhost->vm_stats;
+#else
    const struct vm_statistics* vm = &dhost->vm_stats;
+#endif
    double page_K = (double)vm_page_size / (double)1024;
 
    mtr->total = dhost->host_info.max_mem / 1024;
+#ifdef HAVE_STRUCT_VM_STATISTICS64
+   natural_t used = vm->active_count + vm->inactive_count +
+              vm->speculative_count + vm->wire_count +
+              vm->compressor_page_count - vm->purgeable_count - vm->external_page_count;
+   mtr->values[MEMORY_METER_USED] = (double)(used - vm->compressor_page_count) * page_K;
+#else
    mtr->values[MEMORY_METER_USED] = (double)(vm->active_count + vm->wire_count) * page_K;
+#endif
```

Comment:
Any reasons for this calculation to differ between those two places, besides missing fields from the old struct?

Also, `vm->compressor_page_count` is substracted twice AFAICS.

Reply:
As https://github.com/htop-dev/htop/issues/631 and https://github.com/htop-dev/htop/issues/955 said, the current memory calculation may have a bit problem. 

I referenced the code of  [exelban/stats](https://github.com/exelban/stats/blob/c78319bdecdf56a73b8edfd791f727a3c1c1451c/Modules/RAM/readers.swift#L56) and [GuillaumeGomez/sysinfo](
https://github.com/GuillaumeGomez/sysinfo/blob/9705c681a61f9443b63115b4bc7720e86a3d63ee/src/unix/apple/system.rs#L151
)(used by `ClementTsang/bottom` mentioned by https://github.com/htop-dev/htop/issues/955 ) 

There are some differences in details, but it seems that the main thing missing is adding `vm->compressor_page_count`

It seems that many people have raised the same issue of memory parameter display. 
So I just want macOS to display the correct memory parameters. But there may be something I haven't considered.